docs/user-guide/src/bmo/bmo.md

```diff
@@ -0,0 +1,42 @@
+<h1>Bare Metal Operator</h1>
+
+The Bare Metal Operator (BMO) is a custom Kubernetes controller that reconciles BareMetalHost resources using information provided by Ironic and the Cluster API Provider Metal3 (CAPM3) controller. A simplified information flow path is shown below:
+
+```bash
+CAPM3 ----------> BMO <----------> Ironic <----------> Server
+            BareMetalHost resource
+            - server MACAddress                        - image
+            - server address                           - cloud-init data
+            - server credentials
+```
+
+It would help to use an example to describe what BMO does. There are two operations of interest, getting hardware details of the server and  booting the server with a given image, including user supplied cloud-init data. The BareMetalHost resource contains address and authentication information towards a server.
+
+BMO communicates this information to Ironic and gets hardware details, such as CPU and disk, of the server in return. This information is added to the BareMetalHost resource. If there are hardware changes, the BareMetalHost is updated accordingly.
+
+```bash
+CAPM3 ----------> BMO <----------> Ironic <----------> Server
+            BareMetalHost resource
+            - server MACAddress                        - image
+            - server address                           - cloud-init data
+            - server credentials
+            - Hardware details [cpu,disk,...]
+```
+
+Next, with the information combing from the CAPM3 side, BMO updates the BareMetalHost with image and cloud-init data.
```

Comment:
Maybe add some explanation for why this is coming from CAPM3? For example:
> This happens for example when the user scales a MachineDeployment so that the server should be added to the cluster, or during an upgrade when it must change the image it is booting from.

Reply:
Added the suggested content